### DISCUSSION TITLE
Moving/editing parts of drawing on vector later?

### DISCUSSION THREAD

**Discussion by Baphvincent** [keiwando/toonsquid#170] [Q&A] [answered]
I try to select a part of a drawing and then move/edit it, however it affects the whole layer? This only happens on a vector layer. Is there a way to select parts and edit them like on a pixel layer or is it impossible? I love the vector brushes but moving little parts around to see how they look in different positions is very important to me as a beginner animator. Thanks

**Comment by Cat-Butler** [comment, marked as answer]
Selection, in the meaning you seem to suggest, is a pixel tool.

Selection, in the same meaning of raster/pixel selection, does not exist in a vector context (in ToonSquid or any vector editing app).
In a vector app, you can select objects (an entire curve/shape, with or without fill) or nodes, and that's all. 

In the vector context in ToonSquid, a vector brush drawing layer counts as a single object. 
In ToonSquid, nodes can only be edited if you created the curve/shape with the Path tool, not for content created with a vector brush, alas.

However, you can convert a vector brush layer or a vector path layer into a pixel layer, and then use pixel tools, including selection and translation/rotation. 
Tap on the layer in the drawing layers list and select Rasterize from the pop-up menu.

**Reply by Baphvincent** [reply]
Damn that sucks :/ I'm assuming there's no way to turn it back into a vector layer afterwards? The vector brush just looks so much clearer! Thanks for the response!

**Reply by Cat-Butler** [reply]
Well, you can always keep a hidden duplicate of the original (vector brush, pre-rasterized) layer.

Note: 
For an even better workaround (perhaps), you could simply duplicate the vector brush layer as many times as you need, and (carefully) erase the parts you don't need in each layer, effectually ending up with your original idea of separating out the component parts into separate pieces (drawing layers) i.e. duplicate and erase rather than select (and cut/paste).

You can then happily move or manipulate the individual pieces while retaining their vector brush character.

(The paranoia of experience: Keep a copy of the full original layer, hidden, just in case.)


More crazed mutterings and mumblings:

Weirdly, erase is a vector possibility where selection isn't (although it is a completely solid eraser, "behind the scenes" it is actually more like using a knife or cookie cutter to "nibble" bits off the vector content rather than opacity-based erasing in a pixel sense).
"Cookie cutter" examples, used to cut a shape out of a flat slab of dough: https://www.amazon.com.au/GWHOLE-Cookie-Cutters-Unicorn-Classic/dp/B01MXSVCYQ

In a full-featured vector app, you could "select" (almost like your original idea) by using the same cookie-cutter effect of using one vector shape (the selection) to chop another vector shape (or shapes) apart. 
Go here and scroll down to the "Divide" operation as an example:
https://affinity.help/photo2/en-US.lproj/index.html?page=pages/CurvesShapes/join.html?title=Joining

Unfortunately, this advanced shape interaction is not (currently) available in ToonSquid (and would probably be quite some work to implement. Still, we can wish.).

**Reply by jordanwmartin** [reply]
Selection like you're asking for is actually possible in other vector applications. Most of the vector applications that I use allow this. Adobe Fresco and Animate CC both have vector tools, and allow you to use the Lasso tool to select part of a shape or shapes in the same way you would with pixels in another app. Illustrator CC also always you to do a similar thing with the Knife tool. Illustrator is a little different, but the end result is the same thing.

**Reply by Cat-Butler** [reply]
Hey @jordanwmartin , thanks for giving me a heads up on that - got to play briefly with a Windows version of Fresco recently, and yes it's well done to make vector selections work like pixels.

I personally regard the Knife tool as more of a cookie-cutter-on-the-fly, but I can see your viewpoint too.

And I'm glad Keiwan took the Fresco-like approach and added pixel-style selection to the vector brush layers.

**Reply by jordanwmartin** [reply]
No Problem. Adobe Fresco is easily my favorite drawing application. I agree the Knife tool isn't quite the same, but I also agree that I like the implementation here.

**Comment by keiwando** [comment]
This is now possible since the 1.8 update:
https://youtu.be/ScqSgLppWi8?t=115

**Reply by jordanwmartin** [reply]
I love it, the whole update is really awesome!
